remappers/internal/metric.go

```diff
@@ -70,6 +70,9 @@ func AddMetrics(
 
 		dp.Attributes().PutStr("event.module", common.RemapperEventModule)
 		if dataset != "" {
+			if dataset == "kubernetes.pod" {
+				dp.Attributes().PutStr("service.type", "kubernetes")
+			}
```

Comment:
We can move this logic to the `remappers/kubernetesmetrics/kubelet.go` by using a custom mutator instead of `remappers.EmptyMutator`. The mutator will be something like:

```go
func(dp pmetric.NumberDataPoint) {
	dp.Attributes().PutStr("service.type", "kubernetes")
}
```

Reply:
@lahsivjar thank you done as well. We are currently testing the latest changes and let u know if sth is broken or not

Reply:
Great! Thanks for the changes, one thing remaining is that we are currently not obeying the config `WithKubernetesIntegrationDataset`. I think we can follow an approach similar to https://github.com/elastic/opentelemetry-lib/blob/76a6d08387f09b1372328f5439fdd43a0bfd2670/remappers/hostmetrics/hostmetrics.go#L89-L97 for deducing the dataset based on the config option.

We require to not set dataset for the OTel-Collector -> APM-Server|MIS -> ES path where the dataset would be set by the APM/MIS libraries.

Reply:
I dont think is feasible in k8s case as in hostmetrics is defined based on the scraper that is provided. And it is 1-1 relationship between scraper and dataset
In k8s case this is not valid. I mean we have 2 scrapers (kubeletstatsreceiver and k8sclusterreceiver) but these can save even to same dataset. How about leaving it as it is?

Reply:
> And it is 1-1 relationship between scraper and dataset
> In k8s case this is not valid

Sorry if I am missing something obvious here but to me it looks like the latest code uses the dataset `kubernetes.pod` for `addKubeletMetrics` and `kubernetes.node` for `addClusterMetrics`. If this is true then we can deduce the data set and pass it directly to the remappers.

>  How about leaving it as it is?

I am okay with this too though we would still need to handle `WithKubernetesIntegrationDataset` option where the dataset should be set only if the option is set to `true`, right?

Reply:
@lahsivjar this is accurate at the moment:
addKubeletMetrics -> kubernetes.pod
addClusterMetrics -> kubernetes.node

And actually the clustermetrics are not even used in the UI. But we decided to inlcude them for now as there are some references of those inside Kibana code. 

But in general the kubelet can route to many datastreams. Although for now this is not the case I agree. I just dont want to introduce something now that we might change in the future. Let me also discuss this internally and can come back with you.

In general we have built locally the changes and this code works

Reply:
Got it. Makes sense to not use the same approach as we do for the hostmetrics.

We still need to obey the config option `WithKubernetesIntegrationDataset`, so maybe we can pass the config option to the `addKubeletMetrics` and `addClusterMetrics` where the dataset can be set to empty if the option is `false`.

Reply:
Ok because I see that it raises more confusion for now I copied what is in hostmetrics. 
See https://github.com/elastic/opentelemetry-lib/pull/20#discussion_r1636665657

I think that solves the diffs @lahsivjar